control-plane/plugin/src/resources/node.rs

```diff
@@ -162,47 +183,70 @@ impl Cordoning for Node {
             }
         }
     }
-
-    async fn get_labels(id: &Self::ID, output: &OutputFormat) {
-        match RestClient::client().nodes_api().get_node(id).await {
-            Ok(node) => {
-                let node_display =
-                    NodeDisplay::new(node.into_body(), NodeDisplayFormat::CordonLabels);
-                print_table(output, node_display);
-            }
-            Err(e) => {
-                println!("Failed to get node {}. Error {}", id, e)
-            }
-        }
-    }
 }
 
 /// Display format options for a `Node` object.
 #[derive(Debug)]
-enum NodeDisplayFormat {
+pub enum NodeDisplayFormat {
     Default,
     CordonLabels,
+    Drain,
 }
 
 /// The NodeDisply structure is responsible for controlling the display formatting of Node objects.
 /// `#[serde(flatten)]` and `#[serde(skip)]` attributes are used to ensure that when the object is
 /// serialised, only the `inner` object is represented.
 #[derive(Serialize, Debug)]
-struct NodeDisplay {
+pub struct NodeDisplay {
     #[serde(flatten)]
-    inner: openapi::models::Node,
+    pub inner: Vec<openapi::models::Node>,
     #[serde(skip)]
     format: NodeDisplayFormat,
 }
 
 impl NodeDisplay {
     /// Create a new `NodeDisplay` instance.
     pub(crate) fn new(node: openapi::models::Node, format: NodeDisplayFormat) -> Self {
+        let vec: Vec<openapi::models::Node> = vec![node];
+        Self { inner: vec, format }
+    }
+    pub(crate) fn new_nodes(nodes: Vec<openapi::models::Node>, format: NodeDisplayFormat) -> Self {
         Self {
-            inner: node,
+            inner: nodes,
             format,
         }
     }
+    pub(crate) fn get_label_list(node: &openapi::models::Node) -> Vec<String> {
+        let mut cordon_labels: Vec<String> = vec![];
+        let mut drain_labels: Vec<String> = vec![];
+        if node.spec.is_some() {
+            match &node.spec.as_ref().unwrap().cordondrainstate {
```

Comment:
```suggestion
        if let Some(spec) = &node.spec {
          match spec.cordondrainstate
```
or 
```suggestion
        match node.spec.as_ref().flat_map(|s| s.cordondrainstate) {
         Some(v) =>
         None =>
        }
```

Reply:
I meant to use your 1st suggestion but ended up with 2 layers of 'match'es.

Reply:
fixed